lib/engine/game/g_1822_nrs/game.rb

```diff
@@ -0,0 +1,520 @@
+                'nodes' => ['city'],
+                'pay' => 1,
+                'visit' => 1,
+              },
+              {
+                'nodes' => ['town'],
+                'pay' => 1,
+                'visit' => 1,
+              },
+            ],
+            num: 14,
+            price: 50,
+            rusts_on: '3',
+            variants: [
+              {
+                name: '2',
+                distance: 2,
+                price: 120,
+                rusts_on: '4',
+                available_on: '1',
+              },
+            ],
+          },
+          {
+            name: '3',
+            distance: 3,
+            num: 6,
+            price: 200,
+            rusts_on: '6',
+          },
+          {
+            name: '4',
+            distance: 4,
+            num: 4,
+            price: 300,
+            rusts_on: '7',
+          },
+          {
+            name: '5',
+            distance: 5,
+            num: 2,
+            price: 500,
+            events: [
+              {
+                'type' => 'close_concessions',
+              },
+            ],
+          },
+          {
+            name: '6',
+            distance: 6,
+            num: 3,
+            price: 600,
+            events: [
+              {
+                'type' => 'full_capitalisation',
+              },
+            ],
+          },
+          {
+            name: '7',
+            distance: 7,
+            num: 20,
+            price: 750,
+            variants: [
+              {
+                name: 'E',
+                distance: 99,
+                multiplier: 2,
+                price: 1000,
+              },
+            ],
+            events: [
+              {
+                'type' => 'phase_revenue',
+              },
+            ],
+          },
+          {
+            name: '2P',
+            distance: 2,
+            num: 2,
+            price: 0,
+          },
+          {
+            name: 'LP',
+            distance: [
+              {
+                'nodes' => ['city'],
+                'pay' => 1,
+                'visit' => 1,
+              },
+              {
+                'nodes' => ['town'],
+                'pay' => 1,
+                'visit' => 1,
+              },
+            ],
+            num: 1,
+            price: 0,
+          },
+          {
+            name: '5P',
+            distance: 5,
+            num: 1,
+            price: 0,
+          },
+          {
+            name: 'P+',
+            distance: [
+              {
+                'nodes' => ['city'],
+                'pay' => 99,
+                'visit' => 99,
+              },
+              {
+                'nodes' => ['town'],
+                'pay' => 99,
+                'visit' => 99,
+              },
+            ],
+            num: 2,
+            price: 0,
+          },
+        ].freeze
+
+        UPGRADE_COST_L_TO_2_PHASE_2 = 70
+
+        def discountable_trains_for(corporation)
```

Comment:
I see this function is the same as 1822MRS. Are their rules similar enough that it would make sense for NRS to be a subclass of MRS?

Reply:
Besides the setup differences (different corps, privates, minors), there is one fewer 3 train and some of the tiles are different. The stock market is reused directly. Seems like there are enough differences that it would be best as is? Not sure.

Reply:
Its just that specific method and const that is the same. Everything else is different. Its working great with your new variant as well so i would recommend to keep them as seperate.